client/signup/steps/woocommerce-install/select-site/index.tsx

```diff
@@ -0,0 +1,64 @@
+import { isBusinessPlan } from '@automattic/calypso-products';
+import { Card } from '@automattic/components';
+import { useI18n } from '@wordpress/react-i18n';
+import { ReactElement, useState, useEffect } from 'react';
+import { useDispatch } from 'react-redux';
+import SiteSelector from 'calypso/components/site-selector';
+import StepWrapper from 'calypso/signup/step-wrapper';
+import { submitSignupStep } from 'calypso/state/signup/progress/actions';
+import type { WooCommerceInstallProps } from '../';
+
+import './style.scss';
+
+interface Site {
+	capabilities: {
+		manage_options: boolean;
+	};
+	jetpack: boolean;
+	plan: {
+		product_slug: string;
+	};
+}
+
+export default function SelectSite( props: WooCommerceInstallProps ): ReactElement | null {
+	const { goToStep } = props;
+	const { __ } = useI18n();
+
+	// todo: we should allow any (non jetpack, non p2) site to show up here,
+	// and push people through the checkout step as needed
+	const filterSites = ( site: Site ) => {
+		return (
+			isBusinessPlan( site.plan.product_slug ) && site.capabilities.manage_options && ! site.jetpack
+		);
+	};
+
+	const [ selectedSiteId, setSelectedSiteId ] = useState( '' );
```

Comment:
simple and maybe a naive question. Why do you use an empty string here? can we use simply do not define the initial state... and below into the use effect hook simply check whether the value is defined?

```es6
const [ selectedSiteId, setSelectedSiteId ] = useState( );

// ...

useEffect( () => {
	if ( ! selectedSiteId ) {
		return;
	}

	// ...

}, [ selectedSiteId, goToStep, dispatch ] );
```

Reply:
I think typescript was having a whinge, I also originally thought this was a slug not an id but the component sets an id and this can probably be changed to a number. (I recall adding the "Id" on the end of the variable name)